exercises/rail-fence-cipher/src/RailFenceCipher.hs

```diff
@@ -0,0 +1,7 @@
+module RailFenceCipher (encode, decode) where
+
+encode :: Int -> [a] -> [a]
```

Comment:
I see you ask whether to change this. I would use `String` here simply because that is all tests will use, and it's fine if students realise it can be generalised and therefore change it back to `[a]`. I would therefore also not change the example, minus the comment I make on it.

Reply:
But why make it less general when it works fine with other types?

Reply:
One thing I see is that https://github.com/exercism/problem-specifications/blob/master/exercises/rail-fence-cipher/description.md only talks about working with strings, thus that is the extent of the problem being specified. Therefore, any function that I write that adheres to this specification **need** only operate with strings. It is of course great if it works with other types. I imagine most solutions will. The possible complaint with putting `[a]` here is that "why is this stub asking for more than the problem specification is asking for?" the possible complaint with putting `String` is "this stub is unnecessarily restricting this function to only work with the specification, but it could do more"

It's not a perfect analogue, but there are text like https://github.com/exercism/haskell/blob/master/exercises/leap/HINTS.md or https://github.com/exercism/haskell/blob/master/exercises/series/HINTS.md that we can add.

I don't remember right now whether this track has a policy or other examples of either choice ("give minimal signature that satisfies the specification and let students expand" or "give the most general signature that a reasonable solution should be able to achieve") and I can't go looking for examples now, so I make my recommendation based on what I can see now, but I would change if there is an example of something we have done in the past.

Reply:
Well, I thing with ```Int -> String -> String``` would be just fine, afterall